doc/TAM/SAI-Proposal-TAM-MOD-Localhost.md

```diff
@@ -0,0 +1,440 @@
+
+```c
+
+typedef enum _sai_hostif_user_defined_trap_type_t {
+    ...
+    ...
+
+     /** FDB traps */
+     SAI_HOSTIF_USER_DEFINED_TRAP_TYPE_FDB,
+
+     /** In Segment Entry traps */
+     SAI_HOSTIF_USER_DEFINED_TRAP_TYPE_INSEG_ENTRY,
+
++    /** Traps to be associated with TAM collector */
++    SAI_HOSTIF_USER_DEFINED_TRAP_TYPE_TAM,
++
+     /** Custom range base */
+     SAI_HOSTIF_USER_DEFINED_TRAP_TYPE_CUSTOM_RANGE_BASE = 0x00001000,
+
+     /**
+      * @brief End of user defined trap types
+      */
+     SAI_HOSTIF_USER_DEFINED_TRAP_TYPE_END,
+
+ } sai_hostif_user_defined_trap_type_t;
+
+```
+
+```c
+typedef enum _sai_tam_collector_attr_t {
+    ...
+    ...
+     SAI_TAM_COLLECTOR_ATTR_TRANSPORT,
+
+     /**
+      * @brief DSCP value
+      *
+      * @type sai_uint8_t
+      * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+      */
+     SAI_TAM_COLLECTOR_ATTR_DSCP_VALUE,
+
++    /**
++     * @brief Hostif User Defined Trap object used to reach local host via GENETLINK
++     *
++     * @type sai_object_id_t
++     * @flags CREATE_AND_SET
++     * @objects SAI_OBJECT_TYPE_HOSTIF_USER_DEFINED_TRAP
++     * @allownull true
++     * @default SAI_NULL_OBJECT_ID
++     * @validonly SAI_TAM_COLLECTOR_ATTR_LOCALHOST == true
++     */
++    SAI_TAM_COLLECTOR_ATTR_HOSTIF_TRAP,
++
+     /**
+      * @brief End of Attributes
+      */
+     SAI_TAM_COLLECTOR_ATTR_END,
+
+}  sai_tam_collector_attr_t;
+```
+
+### New Report type for GENETLINK
+
+* The report object is a mandatory attribute to create a TAM Event Action. And to create a report object, the report type is a mandatory attribute. The currently supported report types (see below) are not relevant for GENETLINK. So, a new report type is added for GENETLINK.
+
+* The enum for vendor specific report type (SAI_TAM_REPORT_TYPE_VENDOR_EXTN) does not start after an offset from the standard report types. This can cause new report types to overlap with vendor specific report types. In order to fix this:
+    * Deprecating the current vendor specific report type.
+    * Introducing a custom range base that should be used for vendor specific extensions.
+
+
+
+```c
+typedef enum _sai_tam_report_type_t{
+
+    /**
+     * @brief Report using SFLOW
+     */
+    SAI_TAM_REPORT_TYPE_SFLOW,
+
+    /**
+     * @brief Report using IPFIX
+     */
+    SAI_TAM_REPORT_TYPE_IPFIX,
+
+    /**
+     * @brief Report using GPB
+     */
+    SAI_TAM_REPORT_TYPE_PROTO,
+
+    /**
+     * @brief Report using THRIFT
+     */
+    SAI_TAM_REPORT_TYPE_THRIFT,
+
+    /**
+     * @brief Report using JSON
+     */
+    SAI_TAM_REPORT_TYPE_JSON,
+
+    /**
+     * @brief Report using P4 format
+     */
+     SAI_TAM_REPORT_TYPE_P4_EXTN,
+
+     /**
+      * @brief Report using Histogram
+      */
+     SAI_TAM_REPORT_TYPE_HISTOGRAM,
+
+     /**
+      * @brief Report using vendor extensions
++     * deprecate and use CUSTOM_RANGE_START
+      */
+     SAI_TAM_REPORT_TYPE_VENDOR_EXTN,
```

Comment:
vendor extension is kept as a valid report type instead of using the extension range. This was a specific request during the early TAM review